Doesn't build with GHC-9.0

The `conduit` library doesn't build with `ghc-9.0-rc1`. When trying to build with the new compiler, the following compilation errors occur:

```
[ 2 of 14] Compiling Data.Conduit.Internal.Conduit ( src/Data/Conduit/Internal/Conduit.hs, dist/build/Data/Conduit/Internal/Conduit.o, dist/build/Data/Conduit/Internal/Conduit.dyn_o )

src/Data/Conduit/Internal/Conduit.hs:332:15: error:
    • Couldn't match type: forall b.
                           (() -> Pipe () () o () m b) -> Pipe () () o () m b
                     with: (r4 -> Pipe l4 i4 o3 u3 m3 r4) -> Pipe () i5 o () m ()
      Expected: ConduitT () o m ()
                -> (r4 -> Pipe l4 i4 o3 u3 m3 r4) -> Pipe () i5 o () m ()
        Actual: ConduitT () o m ()
                -> forall b. (() -> Pipe () () o () m b) -> Pipe () () o () m b
    • In the first argument of ‘flip’, namely ‘unConduitT’
      In the second argument of ‘(.)’, namely ‘flip unConduitT Done’
      In the expression: go . flip unConduitT Done
    • Relevant bindings include
        sourceToPipe :: Source m o -> Pipe l i o u m ()
          (bound at src/Data/Conduit/Internal/Conduit.hs:331:1)
    |
332 |     go . flip unConduitT Done
    |               ^^^^^^^^^^

src/Data/Conduit/Internal/Conduit.hs:342:33: error:
    • Couldn't match type: forall b.
                           (r -> Pipe i i Void () m b) -> Pipe i i Void () m b
                     with: (r3 -> Pipe l3 i3 o2 u2 m2 r3) -> Pipe i i Void () m r
      Expected: ConduitT i Void m r
                -> (r3 -> Pipe l3 i3 o2 u2 m2 r3) -> Pipe i i Void () m r
        Actual: ConduitT i Void m r
                -> forall b. (r -> Pipe i i Void () m b) -> Pipe i i Void () m b
    • In the first argument of ‘flip’, namely ‘unConduitT’
      In the second argument of ‘(.)’, namely ‘flip unConduitT Done’
      In the second argument of ‘(.)’, namely
        ‘injectLeftovers . flip unConduitT Done’
    • Relevant bindings include
        sinkToPipe :: Sink i m r -> Pipe l i o u m r
          (bound at src/Data/Conduit/Internal/Conduit.hs:341:1)
    |
342 |     go . injectLeftovers . flip unConduitT Done
    |                                 ^^^^^^^^^^

src/Data/Conduit/Internal/Conduit.hs:352:33: error:
    • Couldn't match type: forall b.
                           (() -> Pipe i i o () m b) -> Pipe i i o () m b
                     with: (r2 -> Pipe l2 i2 o1 u1 m1 r2) -> Pipe i i o () m ()
      Expected: ConduitT i o m ()
                -> (r2 -> Pipe l2 i2 o1 u1 m1 r2) -> Pipe i i o () m ()
        Actual: ConduitT i o m ()
                -> forall b. (() -> Pipe i i o () m b) -> Pipe i i o () m b
    • In the first argument of ‘flip’, namely ‘unConduitT’
      In the second argument of ‘(.)’, namely ‘flip unConduitT Done’
      In the second argument of ‘(.)’, namely
        ‘injectLeftovers . flip unConduitT Done’
    • Relevant bindings include
        conduitToPipe :: Conduit i m o -> Pipe l i o u m ()
          (bound at src/Data/Conduit/Internal/Conduit.hs:351:1)
    |
352 |     go . injectLeftovers . flip unConduitT Done
    |                                 ^^^^^^^^^^

src/Data/Conduit/Internal/Conduit.hs:405:24: error:
    • Couldn't match type: forall b1.
                           (r -> Pipe i i o () m b1) -> Pipe i i o () m b1
                     with: (r -> Pipe i i o () m b) -> Pipe i i o () m b
      Expected: ConduitT i o m r
                -> (r -> Pipe i i o () m b) -> Pipe i i o () m b
        Actual: ConduitT i o m r
                -> forall b. (r -> Pipe i i o () m b) -> Pipe i i o () m b
    • In the first argument of ‘flip’, namely ‘unConduitT’
      In the first argument of ‘(.)’, namely ‘flip unConduitT rest’
      In the second argument of ‘(.)’, namely
        ‘flip unConduitT rest . onErr’
    • Relevant bindings include
        run :: forall a. m a -> IO a
          (bound at src/Data/Conduit/Internal/Conduit.hs:404:44)
        mp :: m (Pipe i i o () m r)
          (bound at src/Data/Conduit/Internal/Conduit.hs:404:15)
        go :: Pipe i i o () m r -> Pipe i i o () m b
          (bound at src/Data/Conduit/Internal/Conduit.hs:403:5)
        rest :: r -> Pipe i i o () m b
          (bound at src/Data/Conduit/Internal/Conduit.hs:402:42)
        onErr :: e -> ConduitT i o m r
          (bound at src/Data/Conduit/Internal/Conduit.hs:402:22)
        p0 :: forall b. (r -> Pipe i i o () m b) -> Pipe i i o () m b
          (bound at src/Data/Conduit/Internal/Conduit.hs:402:18)
        (Some bindings suppressed; use -fmax-relevant-binds=N or -fno-max-relevant-binds)
    |
405 |         (return . flip unConduitT rest . onErr)
    |                        ^^^^^^^^^^

src/Data/Conduit/Internal/Conduit.hs:697:15: error:
    • Couldn't match type: forall b.
                           (() -> Pipe () () a () m b) -> Pipe () () a () m b
                     with: (r0 -> Pipe l0 i0 o0 u0 m0 r0) -> Pipe l1 i1 a () m r1
      Expected: ConduitT () a m ()
                -> (r0 -> Pipe l0 i0 o0 u0 m0 r0) -> Pipe l1 i1 a () m r1
        Actual: ConduitT () a m ()
                -> forall b. (() -> Pipe () () a () m b) -> Pipe () () a () m b
    • In the first argument of ‘flip’, namely ‘unConduitT’
      In the second argument of ‘(.)’, namely ‘flip unConduitT Done’
      In the expression: go . flip unConduitT Done
    • Relevant bindings include
        sourceToList :: Source m a -> m [a]
          (bound at src/Data/Conduit/Internal/Conduit.hs:696:1)
    |
697 |     go . flip unConduitT Done
    |               ^^^^^^^^^^
cabal: Failed to build conduit-1.3.3
```

--- a/conduit/src/Data/Conduit/Internal/Conduit.hs
+++ b/conduit/src/Data/Conduit/Internal/Conduit.hs
@@ -329,7 +329,7 @@ connectResume (SealedConduitT left0) (ConduitT right0) =
 
 sourceToPipe :: Monad m => Source m o -> Pipe l i o u m ()
 sourceToPipe =
-    go . flip unConduitT Done
+    go . (`unConduitT` Done)
   where
     go (HaveOutput p o) = HaveOutput (go p) o
     go (NeedInput _ c) = go $ c ()
@@ -339,7 +339,7 @@ sourceToPipe =
 
 sinkToPipe :: Monad m => Sink i m r -> Pipe l i o u m r
 sinkToPipe =
-    go . injectLeftovers . flip unConduitT Done
+    go . injectLeftovers . (`unConduitT` Done)
   where
     go (HaveOutput _ o) = absurd o
     go (NeedInput p c) = NeedInput (go . p) (const $ go $ c ())
@@ -349,7 +349,7 @@ sinkToPipe =
 
 conduitToPipe :: Monad m => Conduit i m o -> Pipe l i o u m ()
 conduitToPipe =
-    go . injectLeftovers . flip unConduitT Done
+    go . injectLeftovers . (`unConduitT` Done)
   where
     go (HaveOutput p o) = HaveOutput (go p) o
     go (NeedInput p c) = NeedInput (go . p) (const $ go $ c ())
@@ -402,7 +402,7 @@ catchC :: (MonadUnliftIO m, Exception e)
 catchC (ConduitT p0) onErr = ConduitT $ \rest -> let
     go (Done r) = rest r
     go (PipeM mp) = PipeM $ withRunInIO $ \run -> E.catch (run (liftM go mp))
-        (return . flip unConduitT rest . onErr)
+        (return . (`unConduitT`rest) . onErr)
     go (Leftover p i) = Leftover (go p) i
     go (NeedInput x y) = NeedInput (go . x) (go . y)
     go (HaveOutput p o) = HaveOutput (go p) o
@@ -694,7 +694,7 @@ passthroughSink (ConduitT sink0) final = ConduitT $ \rest -> let
 -- Since 1.2.6
 sourceToList :: Monad m => Source m a -> m [a]
 sourceToList =
-    go . flip unConduitT Done
+    go . (`unConduitT` Done)
   where
     go (Done _) = return []
     go (HaveOutput src x) = liftM (x:) (go src)

